### PR TITLE
Construction Blast door

### PR DESCRIPTION
**Pull request by Nimfar11**
## About the PR

Gives you the ability to build blast doors. Disassembly is a little more complicated.

## Why / Balance

Sometimes there is not enough opportunity to build blast doors in place of those destroyed.

## Media

https://github.com/space-wizards/space-station-14/assets/128169402/6ed4dbe9-dfdf-4431-9676-d216d9f3202b

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

:cl: Nimfar11

- add: Adds the ability to build blast doors.

### CODE REVIEW

File: Resources/Prototypes/Recipes/Construction/Graphs/structures/blast_door.yml [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+          conditions:
+            - !type:EntityAnchored
+              anchored: true
+          steps:
+            - tool: Cutting
+              doAfter: 4
+            - tool: Prying
+              doAfter: 2
+
+    - node: frame3
```
Comment (deltanedas) [inline Resources/Prototypes/Recipes/Construction/Graphs/structures/blast_door.yml:66]: need an edge to frame2 so you can door elec -> oops i dont want to anymore without screwing and removing

File: Resources/Prototypes/Recipes/Construction/Graphs/structures/blast_door.yml [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+        - to: BlastDoor
+          conditions:
+            - !type:EntityAnchored
+              anchored: true
+          steps:
+            - tool: Anchoring
+              doAfter: 2
+        - to: frame2
+          completed:
+            - !type:SpawnPrototype
```
Comment (deltanedas) [inline Resources/Prototypes/Recipes/Construction/Graphs/structures/blast_door.yml:87]: should use containers like airlock

File: Resources/Textures/Structures/Doors/Shutters/blastdoor.rsi/meta.json [hunk 3]
```diff
@@ -1 +1,52 @@
-{"name":1,"size":{"x":32,"y":32},"states":[{"name":"closing","directions":1,"delays":[[0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.4]]},{"name":"opening","directions":1,"delays":[[0.1,0.1,0.1,0.1,0.1,0.1,0.1,0.4]]},{"name":"open","directions":1},{"name":"closed","directions":1}],"license":"CC-BY-SA-3.0","copyright":"Tgstation at 97b4295aca9f31a750456e40730d05b5837e39fc","version":1}
\ No newline at end of file
+{
+    "version": 1,
+    "license": "CC-BY-SA-3.0",
+    "copyright": "Tgstation at 97b4295aca9f31a750456e40730d05b5837e39fc",
+    "size": {
+		"x": 32,
```
Comment (deltanedas) [inline Resources/Textures/Structures/Doors/Shutters/blastdoor.rsi/meta.json:6]: the indent is fucked in some lines

File: Resources/Textures/Structures/Doors/Shutters/blastdoor.rsi/meta.json [hunk 4]
```diff
@@ -1 +1,52 @@
...
+                    0.1,
+                    0.1,
+                    0.1,
+                    0.4
+                ]
+            ]
+        },
+        {
+            "name":"opening",
+            "directions":1,
```
Comment (deltanedas) [inline Resources/Textures/Structures/Doors/Shutters/blastdoor.rsi/meta.json:31]: remove

File: Resources/Textures/Structures/Doors/Shutters/blastdoor.rsi/meta.json [hunk 5]
```diff
@@ -1 +1,52 @@
...
+		"x": 32,
+		"y": 32
+    },
+    "states": [
+        {
+            "name": "assembly"
+        },
+        {
+            "name": "closing",
+            "directions": 1,
```
Comment (deltanedas) [inline Resources/Textures/Structures/Doors/Shutters/blastdoor.rsi/meta.json:15]: remove

**Review by deltanedas — CHANGES_REQUESTED** [review]

File: Resources/Prototypes/Recipes/Construction/Graphs/structures/blast_door.yml (hunk 2 above)
Comment (Nimfar11) [reply Resources/Prototypes/Recipes/Construction/Graphs/structures/blast_door.yml]: done

File: Resources/Prototypes/Recipes/Construction/Graphs/structures/blast_door.yml (hunk 1 above)
Comment (Nimfar11) [reply Resources/Prototypes/Recipes/Construction/Graphs/structures/blast_door.yml:66]: I'm not sure I got it right, but I did something )

File: Resources/Textures/Structures/Doors/Shutters/blastdoor.rsi/meta.json (hunk 3 above)
Comment (Nimfar11) [reply Resources/Textures/Structures/Doors/Shutters/blastdoor.rsi/meta.json]: corrected

File: Resources/Prototypes/Recipes/Construction/Graphs/structures/blast_door.yml (hunk 1 above)
Comment (deltanedas) [reply Resources/Prototypes/Recipes/Construction/Graphs/structures/blast_door.yml:66]: frame4 should go to frame3 not frame2, then remove the pry action
frame3 should have a prying action to go back to frame2

Comment (Nimfar11) [reply Resources/Prototypes/Recipes/Construction/Graphs/structures/blast_door.yml:66]: If you go from frame4 to frame3, it breaks the whole deconstruction. everything works fine with frame2.

Comment (Nimfar11) [reply Resources/Prototypes/Recipes/Construction/Graphs/structures/blast_door.yml:66]: disassembly is intentionally more complicated than assembly

Comment (deltanedas) [reply Resources/Prototypes/Recipes/Construction/Graphs/structures/blast_door.yml:66]: ok i guess

File: Resources/Prototypes/Entities/Structures/Doors/Shutter/blast_door.yml [hunk 6]
```diff
@@ -43,3 +53,46 @@
...
+  - type: Destructible
+    thresholds:
+    - trigger:
+        !type:DamageTrigger
+        damage: 300
+      behaviors:
+      - !type:DoActsBehavior
+        acts: ["Destruction"]
+  - type: Physics
+    bodyType: Dynamic
```
Comment (TheShuEd) [inline Resources/Prototypes/Entities/Structures/Doors/Shutter/blast_door.yml:79]: bodyType: Static
Its should fix error checks

Comment (Nimfar11) [reply Resources/Prototypes/Entities/Structures/Doors/Shutter/blast_door.yml:79]: My mistake, I forgot to change it

**Review by TheShuEd — APPROVED** [review]

**Pull request merged**